internal/surgeon/surgeon.go

```diff
@@ -0,0 +1,51 @@
+package surgeon
+
+import (
+	"fmt"
+	"os"
+
+	"go.etcd.io/bbolt/internal/guts_cli"
+)
+
+func CopyPage(path string, srcPage guts_cli.Pgid, target guts_cli.Pgid) error {
+	p1, d1, err1 := guts_cli.ReadPage(path, uint64(srcPage))
+	if err1 != nil {
+		return err1
+	}
+	p1.SetId(target)
+	return WritePage(path, d1)
+}
+
+func WritePage(path string, pageBuf []byte) error {
+	page := guts_cli.LoadPage(pageBuf)
+	pageSize, _, err := guts_cli.ReadPageAndHWMSize(path)
+	if err != nil {
+		return err
+	}
+	if pageSize != uint64(len(pageBuf)) {
+		return fmt.Errorf("WritePage: len(buf)=%d != pageSize=%d", len(pageBuf), pageSize)
+	}
+	f, err := os.OpenFile(path, os.O_WRONLY, 0)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	_, err = f.WriteAt(pageBuf, int64(page.Id())*int64(pageSize))
+	return err
+}
+
+// RevertMetaPage replaces the newer metadata page with the older.
+// It usually means that one transaction is being lost. But frequently
+// data corruption happens on the last transaction pages and the
+// previous state is consistent.
+func RevertMetaPage(path string) error {
```

Comment:
Do you have a use case of when to use this function? 

Probably we do not need this function, because boltDB already supports load the correct meta page. See https://github.com/etcd-io/bbolt/blob/b654ce922133ff49bb385297f30835ca357bac5f/db.go#L1029-L1039

Reply:
I'm planning to expose it as `bbolt surge` command (see https://github.com/etcd-io/bbolt/issues/370). 

The reason is not a corrupted 'meta' page. 
The reason is the last transaction's write that did something bad to the tree structure... so all the data from the current root cannot be reached... but if we traversed the previous root... everything would work.  I've seen 2 such corruptions recently... so I think it's a good tool to have for data-recovery cases.

Reply:
> The reason is not a corrupted 'meta' page.
> The reason is the last transaction's write that did something bad to the tree structure...

Makes some sense to me. The concern is how can a user (especially a normal user) know `last transaction's write that did something bad to the tree structure`?

Reply:
The naive process would be: 
```
   bbolt check db .... -> claims fatal error
   bbolt surgery revert-meta-page. db db.outputFile
   bbolt check db.outputFile .... -> claims success
```
So we can recover the DB at the price of a single transaction. But it might not work as well.

We could integrate it in one experience later.